lib/src/core/provider/calendar_scroll_provider.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:developer';
+
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+class CalendarScrollProvider with ChangeNotifier {
+
+  CalendarScrollProvider(){
+    var currentMonth = DateTime.now();
+    var formatter = DateFormat('LLLL');
+    selectedMonth =  formatter.format(currentMonth);
+    indexedDates = getIndexedDates ();
+  }
+
+  FixedExtentScrollController fixedExtentScrollController;
+  int currentItem = 20;
+  int numberAmount = 61;
+  int indexStartsFrom = 0;
+  String selectedMonth;
+  List<Map<String,  dynamic>> indexedDates;
+
+
+
+  List<Map<String,  dynamic>> getIndexedDates (){
+
+    //selectedMonth =
+
+    var indexedDatesList = List<Map<String,  dynamic>>(numberAmount);
+    for (var index = 0; index <=numberAmount-1; index++ ){
+      var date = DateTime(DateTime.now().year, DateTime.now().month, DateTime.now().day).add(Duration( days: index-20));
```

Comment:
calling `now` constructor every time is relatively expensive. in case we call this every time we create this provider - we can make a variable that stores such information and use it later on